### PR TITLE
Server setup

### PR DESCRIPTION
**Pull request by adisri**
Added a basic build script for the project and used Boost libraries to set up a TCP connection, get a raw request, and send that request back to the client (NOT yet formatted as a response). Tested with curl and Firefox.

### CODE REVIEW

File: main.cpp [hunk 1]
```diff
@@ -0,0 +1,49 @@
+#include <iostream>
+#include <string>
+#include <exception>
+#include <boost/asio.hpp>
+// #include "nginx-configparser/config_parser.h"
```
Comment (u3amar) [inline main.cpp]: Remove commented code

File: main.cpp [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+// Inspired by https://github.com/egalli64/thisthread/blob/master/asio/tcpIpCs.cpp
+void runTcpServer(int port_number) {
+	boost::asio::io_service aios;
+	boost::asio::ip::tcp::acceptor acceptor(aios, boost::asio::ip::tcp::endpoint(boost::asio::ip::tcp::v4(), port_number));
+
+	while (true) {
+    	boost::asio::ip::tcp::socket socket(aios);
```
Comment (u3amar) [inline main.cpp]: Formatting - indent

File: main.cpp [hunk 3]
```diff
@@ -0,0 +1,49 @@
+#include <iostream>
+#include <string>
+#include <exception>
+#include <boost/asio.hpp>
+// #include "nginx-configparser/config_parser.h"
+
+// Inspired by https://github.com/egalli64/thisthread/blob/master/asio/tcpIpCs.cpp
+void runTcpServer(int port_number) {
+	boost::asio::io_service aios;
+	boost::asio::ip::tcp::acceptor acceptor(aios, boost::asio::ip::tcp::endpoint(boost::asio::ip::tcp::v4(), port_number));
```
Comment (u3amar) [inline main.cpp]: Extract method parameters into separate variables (line is too long)

File: main.cpp [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+    	std::stringstream message_stream;
+  		boost::asio::streambuf buffer;
+  		boost::system::error_code error;
+  		size_t len = read_until(socket, buffer, "\r\n\r\n", error);
+  	  
+  		// TODO: format the response with a status line, content-type, etc.
+
+  		if (len) {
+	    	message_stream.write(boost::asio::buffer_cast<const char *>(buffer.data()), len);
```
Comment (u3amar) [inline main.cpp]: Formatting

File: main.cpp [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+	boost::asio::ip::tcp::acceptor acceptor(aios, boost::asio::ip::tcp::endpoint(boost::asio::ip::tcp::v4(), port_number));
+
+	while (true) {
+    	boost::asio::ip::tcp::socket socket(aios);
+    	acceptor.accept(socket);
+
+    	std::stringstream message_stream;
+  		boost::asio::streambuf buffer;
+  		boost::system::error_code error;
+  		size_t len = read_until(socket, buffer, "\r\n\r\n", error);
```
Comment (u3amar) [inline main.cpp]: Extract \r\n\r\n into a const variable

**Review by u3amar — COMMENTED** [review]
Nice work! Watch out for formatting issues.

File: main.cpp (hunk 1 above)
Comment (adisri) [reply main.cpp]: Ok. You'll need to include that again when you link up the config parser.

File: main.cpp (hunk 3 above)
Comment (adisri) [reply main.cpp]: Docs online seem to show that this is the standard way of invoking the boost acceptor method, I think we can keep it this way. I'll reformat each parameter on a new line for readability.

File: main.cpp (hunk 5 above)
Comment (adisri) [reply main.cpp]: Good call.

**Pull request merged**